npm/vite-plugin-cypress-esm/src/index.ts

```diff
@@ -120,21 +118,30 @@ export const CypressEsm = (options?: CypressEsmOptions): Plugin => {
 
     // Ensure import comes at start of line *or* is prefixed by a space so we don't capture things like
     // `Refresh.__hmr_import('')
-    const importRegex = /(?<=^|\s)import (.+?) from (.*)/g
+    const importRegex = /(?<=^|\s)import (.+?) from (['"].*)/g
 
     return code.replace(
       importRegex,
       (match, importVars: string, importTarget: string) => {
-        if (isImportOnIgnoreList(importTarget)) {
-          debug(`⏭️ Import ${importTarget} matches ignoreImportList, ignoring`)
+        // Strip quotes & semicolons
+        const sanitizedImportTarget = importTarget.replace(/["';]/gi, '').trim()
+
+        if (isValidPath(sanitizedImportTarget)) {
+          debug(`🚫 Invalid import target (${sanitizedImportTarget}), ignoring`)
```

Comment:
We saw some import statements in comments or console outputs (see example below) which were picked up by this plugin. Now, each import target has to be a valid path.

Example
```
console.warn(['MUI: `ClassNameGenerator` import from `@mui/material/utils` is outdated (...)'].join('\n'));
```

Reply:
Unfortunately, this doesn't work. I have given up for now to detect import statements inside strings and comments. We will need to strip out this line independently.